src/parser/ir/ir_parser_test.cc

```diff
@@ -25,8 +25,9 @@ class IrParserTest : public testing::TestWithParam<absl::string_view> {};
 
 TEST_P(IrParserTest, SimpleTestOperation) {
   auto input_program_text = GetParam();
-  IrProgramParser ir_parser;
-  EXPECT_EQ(IRPrinter::ToString(ir_parser.ParseProgram(input_program_text)),
+  auto [context, module, ssa_names] =
+      IrProgramParser::ParseProgram(input_program_text);
+  EXPECT_EQ(IRPrinter::ToString(*module, std::move(*ssa_names)),
```

Comment:
See other replies about why `unique_ptr` was  used. 

I agree `std::move` is probably not a clean solution here, but we are limited by the way `SsaNames` are used in the `IrPrinter`. I think we should really move away from treating `SsaNames` as free standing and  make it part of something like `IRContext` or a top-level module.  In the interim, we will need to have a way of passing `ssa_names` to IrPrinter. Perhaps, a _slightly_ cleaner solution is the following:

```
  EXPECT_EQ(IRPrinter::ToString(*module, std::move(*(ssa_names.release()))),
```

Another option is to make `IrPrinter::ToString` to take `std::unique_ptr<SsaNames>`. Even then you will still need to std::move.

I would suggest that we get this in as we need to unblock @asejfia to make progress on the DP project. After getting this in, let us take a step back and rethink where `SsaNames` should be kept.

Reply:
OK, that is fine for me for now. But rather than having a pattern that looks so dangerous, I think it might make sense to eliminate the `std::move` and just make this a copy. It's one copy in a test, so it's not like it will cause performance issues, but it will require less reasoning to understand that it's safe.

Reply:
Using this solution here `EXPECT_EQ(IRPrinter::ToString(*module, std::move(*(ssa_names.release()))),`. Cannot make a copy as SsaNames do not have a copy constructor

Reply:
OK, it's fine that we need to keep the move. But we should not be using `release`. Doing so will cause the memory managed by `std::unique_ptr` to not be freed and will cause a memory leak. This can be safely replaced with just:

```C++
EXPECT_EQ(IRPrinter::ToString(*module, std::move(*(ssa_names)))
```

That will move the value out of `*ssa_names`, but leave the pointer to `ssa_names` in `std::unique_ptr`, allowing the allocated memory to be cleaned up.

Reply:
Mark, I made changes to IRPrinter to hold on to ssa_names as a unique_ptr. Hence, we do not need to have `EXPECT_EQ(IRPrinter::ToString(*module, std::move(*(ssa_names)))` instead it become `EXPECT_EQ(IRPrinter::ToString(*module, std::move((ssa_names)))`

Reply:
ah, ok, sounds good. I was refreshing, but I still saw release. I guess we can chalk that up to a stale view or something. Sorry about that.